pkg/connector/service.go

```diff
@@ -147,6 +148,7 @@ func (s *Service) Update(ctx context.Context, id string, data Config) (Connector
 	}
 
 	conn.SetConfig(data)
+	conn.SetUpdatedAt(time.Now())
```

Comment:
Maybe it's easier if we pull SetUpdatedAt into SetConfig.

Reply:
I almost posted the same comment in the previous review, but after giving it a second thought decided against it. My reasoning - given that the decision to update the `UpdatedAt` field is made in `pipeline.Service` and `processor.Service`, I would rather let `connector.Service` explicitly set it as well, to have some consistency between services.

Reply:
Consistency is a good point. In this case, it appears to be consistency in implementation details of independent services, not about behavior or user experience, so at least IMHO it's still more valuable to pull it into SetConfig, because that has the value of making sure it always gets updated (and we're already calling it multiple times).

Reply:
I'm not convinced tbh, it's a question of "whose responsibility is it to detect an update" and I think it should be the services (at least that's what we do in other services). The fact that we have functions on the connector that could take over this responsibility is only a consequence of the design, which I already [mentioned](https://github.com/ConduitIO/conduit/pull/381#discussion_r864842440) is bad and should change.

Reply:
At least to me, it's the object itself which knows most about itself, including these changes and which should track its own metadata. It might help with race conditions too, and making sure invalid timestamps are not used. That would of course mandate using functions to update fields, so if that's not Go-idiomatic, it makes it less desirable as a solution. However, I certainly don't think this is a blocker.

Reply:
I understand your view and if we were doing OOP I would be on your side! I think the difference in opinions stems from how we think about the connector - you just called it an "object", while I think of it as more of a "dumb struct" that carries data and can be manipulated by other "objects". Of course, my view of the connector breaks down once the pipeline starts running because the "dumb struct" then manages its own state 🙃  So my view depends on the context, and once again it boils down to bad design (I am to blame for that).

Maybe we could make a better distinction between the connector "data struct" (only carries fields that are persisted) and a "running connector" (contains logic for communicating with the plugin), which would make the responsibilities clearer. I really hope we will have some time before the end of the milestone to refactor!